File: hw2/b.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

const mappings = {
  bauk: 'dos',
  flatbaka: 'petsa',
  k: 'g',
  y: 'u'
};

rl.on('line', (input) => {
  let res = input.replace(/bauk|flatbaka|k|y/g, function(matched) {
    return mappings[matched];
  });

  console.log(res);
  process.exit(0);
});

File: hw2/b.java
import java.util.*;

public class b {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String input = in.nextLine();

        in.close();

        String output = input
                .replace("flatbaka", "petsa")
                .replace("bauk", "dos")
                .replace("k", "g")
                .replace("y", "u");

        System.out.println(output);
    }
}
